src/Catalog/Dnx/DnxCatalogCollector.cs

```diff
@@ -70,7 +70,8 @@ public class DnxCatalogCollector : CommitCollector
             var destinationStorage = _storageFactory.Create(id);
             var destinationUri = destinationStorage.GetUri(DnxMaker.GetRelativeAddressNupkg(id, version));
 
-            if (await destinationStorage.AreSyncronized(sourceUri, destinationUri))
+            if (await destinationStorage.AreSyncronized(sourceUri, destinationUri)
+                && await _dnxMaker.HasPackageInIndex(destinationStorage, id, version, cancellationToken))
```

Comment:
I think we can further optimize and simply call into the `UpdateMetadata` only as needed, rather than re-download the nupkg and nuspec which don't need an update. Store the `AreSynchronized` result and skip `nupkg` update based on this value in the `AddPackage`, you can refactor the `AddPackage` method for this if needed. I think it would be a good optimization to have, given we have seen so many incidents around this code path.

Reply:
> I think it would be a good optimization to have, given we have seen so many incidents around this code path.

I'm not sure if I follow. The only time this optimization will make a difference is when then .nupkg is there but the version is not in the index.json. This has only ever happened once in PROD so I'd prefer to keep the code simpler.

Reply:
Oh, I meant the downloads timing out for large size package. If the issue for failure to update index.json were to happen for a large sized package, we would essentially be spending another 10-15 mins for downloading a nupkg which doesn't need to be updated.

Reply:
Yeah, that's a good point. If we can save this, we should. I'll fix this up.

Reply:
Fixed.